### DISCUSSION TITLE
Samsung TV Hue Sync App Integration

### DISCUSSION THREAD

**Discussion by hobointhecorner** [mvdwetering/huesyncbox#53] [General]
I'm using the recently-released Samsung TV Hue Sync app, and I was wondering if there were plans to support it as a hue sync device.

So far, I've found:
- The Hue Sync mobile app doesn't currently recognize the TV app as a sync box
    - The TV app does not attempt, nor does it offer any additional functionality, to pair over bluetooth
    - The mainline Hue app is aware that Sync is enabled in the specific entertainment zone, but the only available option is to stop syncing
- Home Assistant does not automatically discover the TV app as a sync box
- As far as I'm aware, manually adding the integration is not possible, as the `identifier` field isn't available in the TV app

I'm more than willing to provide any additional info or help test, if desired. Unfortunately, my Python skills are incredibly lacking, so I'd need a somewhat substantial amount of guidance to post a somewhat-respectable pull request.

**Comment by mvdwetering** [comment]
The real question is if the TV sync app has an API (hopefully similar to the box). Since the official Hue apps don't support it there probably is no API right now, but if there is I could (probably) add support.

> The mainline Hue app is aware that Sync is enabled in the specific entertainment zone, but the only available option is to stop syncing

That information comes from the bridge, so not related to whatever application is the entertainment source.

You might want to poke Philips Hue on the socials and ask why it is not recognized/controllable by the mobile sync app. Or ask on the Hue developers forums https://developers.meethue.com/ for an API.

Just for my understanding as I don't have a Samsung TV. 
If you want to change some sync settings do you have to close current App (e.g. Netflix), go to Hue Sync TV app, change setting and go back and resume? Or is there some kind of way to easily call up the Hus Sync TV app and swap back? I can imagine that it is a lot more convenient to use a mobile app to change settings while watching (or use Home Assistant to automate it all)

**Comment by hobointhecorner** [comment]
> You might want to poke Philips Hue on the socials and ask why it is not recognized/controllable by the mobile sync app. Or ask on the Hue developers forums https://developers.meethue.com/ for an API.

I'll try to throw something together and post results.

> If you want to change some sync settings do you have to close current App (e.g. Netflix), go to Hue Sync TV app, change setting and go back and resume? Or is there some kind of way to easily call up the Hus Sync TV app and swap back? I can imagine that it is a lot more convenient to use a mobile app to change settings while watching (or use Home Assistant to automate it all)

There's an option to automatically start syncing when the TV turns on, but to adjust the sync settings, you do have to swap apps.

**Reply by hobointhecorner** [reply]
Did a little poking, and it doesn't look like it advertises over mDNS (I can't find the `dns-sd` download for windows, so I'm using `FindDevice.exe`).  I'll definitely have to prod the dev forums to see what, if any, plans they might have to support it via the API.

**Comment by terveer** [comment]
Did you find out anything?
I've tried the bridge's API (as the Hue app can stop the Entertain area).
However a rest request with put on the corresponding group like this: 
`{"stream": {"active": false, "owner": ""}}`
will just result in an error like: 
`[
	{
		"error": {
			"type": 6,
			"address": "/groups/2/action/stream",
			"description": "parameter, stream, not available"
		}
	}
]`

Most likely because the group has an owner ID restricting other apps and scenes from changing the gradient lightstrip's settings.

**Reply by mvdwetering** [reply]
The "owner" part should should not be there.

Without it the command works fine when I send it to the active entertainment area.

```
{"stream":{"active":false}}
```

**Reply by terveer** [reply]
This was my first try actually. It leads to the same error message. Supplying an owner was just another try suspecting the owner might be the culprit. 
I am using a V2 Bridge with Software 1.67.1967054020, Gradient Lightstrip 100b-118 and Samsung The Frame 75 (2023) / GQ75LS03BGUXZG. 

Any idea what is different to your setup and why I wouldn't work for me?

**Reply by mvdwetering** [reply]
Are you sure you are addressing the correct group?

 You have to address the active entertainment group to be able to disable it. 
The group has `"type": "Entertainment",` and a block with stream settings like below.

```
    "stream": {
        "proxymode": "auto",
        "proxynode": "/bridge",
        "active": true,
        "owner": "5e4475c0-502f-4a2e-8df0-b629b3161e08"
    },
```

**Reply by mvdwetering** [reply]
Ow, and just noticed after posting. 

The path should be just `/groups/2` not `/groups/2/action` which you seem to be trying to address, judging from the error message.

**Reply by terveer** [reply]
Thank you so much - this turned out to be the issue! Without /action it's working as intended now.

**Comment by vishal1082** [comment]
This is now also available in newer LG TVs (2024 and later). Is there a plan to add this feature? As I understand, it is only possible to stop the sync but not start it.

**Reply by mvdwetering** [reply]
The stopping of syncing in the way discussed above would be feature request for the Home Assistant built in Hue integration because that is something that is done from by the bridge. Basically you can stop syncing in an entertainment area independent of what is syncing.

The Hue integration already shows the status of an entertainment area. These sensors are part of the bridge device, so I guess it could be an extension for that.
![image](https://github.com/user-attachments/assets/1c7d9a69-c21a-45b4-8fe8-e322f9c1d640)



In the earlier discussion on stopping that person made a REST request directly through the bridge by using the API as described on https://developers.meethue.com which will take some fiddling to get right.
